meta-balena-common/recipes-containers/docker-disk/files/balena-api.inc

```diff
@@ -0,0 +1,283 @@
+#!/bin/bash
+
+CURL="curl -s"
+
+TRANSLATION=${TRANSLATION:-"v6"}
+
+DEBUG=0
+pp_json() {
+	if [ "${DEBUG}" = "1" ]; then
+		echo "${1}" | >&2 jq .
+	fi
+}
+
+dlog() {
+	if [ "${DEBUG}" = "1" ]; then
+		>&2 echo "${1}"
+	fi
+}
+
+# Returns success if the check fails
+check_fail() {
+	local _json
+	local _msg
+	_json="$1"
+	_msg="$2"
+
+	if [ "${_json}" != "OK" ]; then
+		pp_json "${_json}"
+		>&2 echo "${_msg}"
+		return 0
+	fi
+	return 1
+}
+
+# Print public application ID from application name
+# Arguments:
+#
+# $1: Application name
+# $2: Balena target environment
+#
+# Result:
+# Prints the application ID or null if it does not exist
+print_appID_from_appName() {
+	local _appName="$1"
+	local _apiEnv="$2"
+	[ -z "${_appName}" ] && >&2 echo "Application name is required" && return 1
+	local _appID=""
+	local _json=""
+	_json=$(${CURL} -XGET "https://api.${_apiEnv}/${TRANSLATION}/application?\$filter=(app_name%20eq%20'${_appName}')%20and%20(is_public%20eq%20true)" -H "Content-Type: application/json")
+	pp_json "${_json}"
+	_appID=$(echo "${_json}" | jq --raw-output '.d[0].id')
+	dlog "[${_appName}] Application ID is ${_appID}"
+	echo "${_appID}"
+}
+
+# Print UUID from public application name
+# Arguments:
+#
+# $1: Application name
+# $2: Balena target environment
+#
+# Result:
+# Prints the application UUID or nothing if it does not exist
+print_appUUID_from_appName() {
+	local _appName="$1"
+	local _apiEnv="$2"
+	[ -z "${_appName}" ] && >&2 echo "Application name is required" && return 1
+	local _appID=""
+	local _json=""
+	_json=$(${CURL} -XGET "https://api.${_apiEnv}/resin/application?\$filter=(app_name%20eq%20'${_appName}')%20and%20(is_public%20eq%20true)" -H "Content-Type: application/json")
+	pp_json "${_json}"
+	_appUUID=$(echo "${_json}" | jq --raw-output ".d[0].uuid")
+	dlog "[${_appName}] Application UUID is ${_appUUID}"
+	echo "${_appUUID}"
+}
+
+# Count all releases at a specific version and optional variant
+# Arguments:
+#
+# $1: Public application name
+# $2: Balena target environment
+# $3: Release version label
+# $4: Release variant label
+#
+# Result:
+# Prints the releases count metadata in JSON format
+count_releases_json() {
+	local _appName="$1"
+	local _apiEnv="$2"
+	local _version="$3"
+	local _variant="$4"
+	local _json=""
+	local _selector=""
+	local _selector_version=""
+	local _selector_variant=""
+
+	# Get application ID from application name
+	_appID=$(print_appID_from_appName "${_appName}" "${_apiEnv}")
+
+	_selector_version="%20and%20(release_tag/any(rt:(rt/tag_key%20eq%20'version')%20and%20(rt/value%20eq%20%27${_version}%27)))"
```

Comment:
currently using `release_tag`s are a bit of a hack, and we plan to migrate to `release_version` which is guaranteed-unique within an app. the supervisor is already doing this, and i'd think other hostapp extensions/system apps should follow suit.

Reply:
@xginn8 would you have an example query using `release_version`? It was painful to make this query work so I'd appreciate any example.

Reply:
No problem. If I wanted something like a supervisor app at version `12.2.11`, the following would suffice:

```
❯ curl -qs "https://api.balena-cloud.com/v6/release?\$filter=(release_version%20eq%20'v12.2.11')%20and%20(belongs_to__application/any(a:(startswith(a/slug,'balena_os%2F')%20and%20endswith(a/slug,'-supervisor'))%20and%20(a/is_public%20eq%20true)%20and%20(a/is_host%20eq%20false)))" | jq '.d | length'
5
```

if that doesn't help clear up the query, i can either help more or show you easier ways to build queries that i've learned :)

Reply:
Thanks @xginn8. This change is more convoluted as we also need to change the way hostapp extensions are published by the jenkins builders and probably should be made both for hostapps and extensions. I will separate this into a different issue and PR.